api/envoy/extensions/filters/http/rbac/v3/upstream_ip_matcher.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.rbac.v3;
+
+import "envoy/config/core/v3/address.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.rbac.v3";
+option java_outer_classname = "UpstreamIpMatcherProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: RBAC Upstream Ip matcher plugin]
+
+// [#alpha:]
+// [#extension: envoy.filters.http.rbac.matcher.upstream_ip]
+// This is configuration for matching upstream ip.
+message UpstreamIpMatcher {
+  // A CIDR block that will be used to match the upstream IP. This field is only applicable if the
+  // request is routed by dynamic forward proxy. If this field is present and the request did not
```

Comment:
The 2nd sentence is not true. It's applicable if any filter sets the right filter state. I would document it as such, and mention that DFP is an example of a filter that does this when configured to do so (see my other comment about runtime)